ndcube/tests/test_ndcube.py

```diff
@@ -1,819 +1,277 @@
+                             ("ndcube_3d_ln_lt_l", np.s_[1, 1, 1:2]),
+                             ("ndcube_4d_ln_lt_l_t", np.s_[0, 0, :, 0]),
+                             ("ndcube_4d_ln_lt_l_t", np.s_[0, 0, ..., 0]),
+                             ("ndcube_4d_ln_lt_l_t", np.s_[1, 1, 1:2, 1]),
+                         ),
+                         indirect=("ndc",))
+def test_slicing_wave(ndc, item):
+    sndc = ndc[item]
+    assert len(sndc.dimensions) == 1
+    assert set(sndc.wcs.world_axis_physical_types) == {"em.wl"}
+    if sndc.uncertainty is not None:
+        assert np.allclose(sndc.data, sndc.uncertainty.array)
+    if sndc.mask is not None:
+        assert np.allclose(sndc.data > 0, sndc.mask)
+
+    if ndc.extra_coords and ndc.extra_coords.keys():
+        ec = sndc.extra_coords
+        assert set(ec.keys()) == {"bye", "hello", "time"}
+
+    wcs = sndc.wcs
+    assert isinstance(wcs, BaseHighLevelWCS)
+    assert isinstance(wcs.low_level_wcs, SlicedLowLevelWCS)
+    assert wcs.pixel_n_dim == 1
+    assert wcs.world_n_dim == 1
+    assert np.allclose(wcs.array_shape, sndc.data.shape)
+    assert np.allclose(sndc.wcs.axis_correlation_matrix, np.ones(1, dtype=bool))
+
+
+@pytest.mark.parametrize("ndc, item",
+                         (
+                             ("ndcube_3d_ln_lt_l", np.s_[0, :, :]),
+                             ("ndcube_3d_ln_lt_l", np.s_[0, ...]),
+                             ("ndcube_3d_ln_lt_l", np.s_[1, 1:2]),
+                             ("ndcube_3d_ln_lt_l", np.s_[0, :, :]),
+                             ("ndcube_3d_ln_lt_l", np.s_[0, ...]),
+                             ("ndcube_3d_ln_lt_l", np.s_[1, :, 1:2]),
+                             ("ndcube_4d_ln_lt_l_t", np.s_[0, :, :, 0]),
+                             ("ndcube_4d_ln_lt_l_t", np.s_[0, ..., 0]),
+                             ("ndcube_4d_ln_lt_l_t", np.s_[1, 1:2, 1:2, 1]),
+                         ),
+                         indirect=("ndc",))
+def test_slicing_split_celestial(ndc, item):
+    sndc = ndc[item]
+    assert len(sndc.dimensions) == 2
+    if sndc.uncertainty is not None:
+        assert np.allclose(sndc.data, sndc.uncertainty.array)
+    if sndc.mask is not None:
+        assert np.allclose(sndc.data > 0, sndc.mask)
+
+    if ndc.extra_coords and ndc.extra_coords.keys():
+        ec = sndc.extra_coords
+        assert set(ec.keys()) == {"hello", "bye", "time"}
+
+    assert isinstance(sndc.wcs, BaseHighLevelWCS)
+    assert isinstance(sndc.wcs.low_level_wcs, SlicedLowLevelWCS)
+    wcs = sndc.wcs
+    assert wcs.pixel_n_dim == 2
+    assert wcs.world_n_dim == 3
+    assert np.allclose(wcs.array_shape, sndc.data.shape)
+    assert set(wcs.world_axis_physical_types) == {"custom:pos.helioprojective.lat",
+                                                  "custom:pos.helioprojective.lon",
+                                                  "em.wl"}
+    assert np.allclose(wcs.axis_correlation_matrix, np.array([[True, False],
```

Comment:
Sure this should be a hard assert, rather than `allclose`?

Reply:
Don't think it matters much.